src/vm/node_modules/vmload/vmload-datasets.js

```diff
@@ -337,7 +350,11 @@ function zfs(options, cmd, args, lineHandler, callback)
 
     // we don't expect data on stderr, so treat as a warning
     zfs_child.stderr.on('data', function (data) {
-        log.warn({stderr: data.toString()}, 'zfs[' + zfs_child.pid + '] wrote '
+        var str = data.toString();
+        if (str.indexOf('dataset does not exist') !== -1) {
```

Comment:
If you're going to interpret the `stderr` output of `zfs(1M)`, we should sanitise the environment and force the C locale.  For example: [tools/lib/oscmds.js](https://github.com/joyent/sdc-headnode/blob/master/tools/lib/oscmds.js#L28-L57) in `sdc-headnode`.

Interpreting the value passed to a single "`data`" event is also not the right way to consume a stream.  You have to assume that the string you're looking for may straddle multiple events.  You should accumulate the data until a line termination, or the "`end`" event of the stream.